src/main/java/com/endava/endabank/controller/AuthenticationController.java

```diff
@@ -0,0 +1,36 @@
+package com.endava.endabank.controller;
+
+import com.endava.endabank.constants.Routes;
+import com.endava.endabank.dto.user.AuthenticationDto;
+import com.endava.endabank.exceptions.customexceptions.BadDataException;
+import com.endava.endabank.service.impl.UserAuthenticationService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.Map;
+
+@RestController
+@RequestMapping(Routes.API_ROUTE)
+@AllArgsConstructor
+public class AuthenticationController {
+    private UserAuthenticationService userAuthenticationService;
+    private AuthenticationManager authenticationManager;
+
+    @PostMapping(value = Routes.LOGIN_ROUTE)
+    public ResponseEntity<Map<String, Object>> createAuthenticationToken(@Valid @RequestBody AuthenticationDto authenticationDto) throws BadDataException {
+        Authentication authentication = authenticate(authenticationDto.getEmail(), authenticationDto.getPassword());
+        return ResponseEntity.ok(userAuthenticationService.logInUser(authentication));
```

Comment:
Hi,

What if somethings fail when "userAuthenticationService.logInUser(authentication)" is executed?

I think you should have a try-catch block to handle exceptions and throw custom exceptions instead of technical ones like BadDataException

if you put the return entity always with ok you are gonna be handling always everything as ok 

Please fix it in the same way for every controller class you have and its methods

Reply:
Hi,
BadData is a custom Exception but i think we don't even need it right there so we will remove it